src/HOCON/HoconParser.cs

```diff
@@ -0,0 +1,1011 @@
+    public delegate Task<string> HoconIncludeCallbackAsync(HoconCallbackType callbackType, string value);
+
+    /// <summary>
+    /// This class contains methods used to parse HOCON (Human-Optimized Config Object Notation)
+    /// configuration strings.
+    /// </summary>
+    public class HoconParser
+    {
+        private readonly List<HoconSubstitution> _substitutions = new List<HoconSubstitution>();
+        private HoconIncludeCallbackAsync _includeCallback = (type, value) => Task.FromResult("{}");
+
+        private HoconTokenizerResult _tokens;
+        private HoconValue _root;
+
+        private HoconPath Path { get; } = new HoconPath();
+
+        /// <summary>
+        /// Parses the supplied HOCON configuration string into a root element.
+        /// </summary>
+        /// <param name="text">The string that contains a HOCON configuration string.</param>
+        /// <param name="includeCallback">Callback used to resolve includes</param>
+        /// <returns>The root element created from the supplied HOCON configuration string.</returns>
+        /// <exception cref="HoconParserException">
+        /// This exception is thrown when an unresolved substitution is encountered.
+        /// It also occurs when any error is encountered while tokenizing or parsing the configuration string.
+        /// </exception>
+        public static HoconRoot Parse(string text, HoconIncludeCallback includeCallback = null)
+        {
+            return new HoconParser().ParseText(text, true, includeCallback);
+        }
+
+        /// <inheritdoc cref="Parse(string, HoconIncludeCallback)"/>
+        public static async Task<HoconRoot> ParseAsync(string text, HoconIncludeCallbackAsync includeCallback = null)
+        {
+            return await new HoconParser().ParseTextAsync(text, true, includeCallback).ConfigureAwait(false);
+        }
+
+        private HoconRoot ParseText(string text, bool resolveSubstitutions, HoconIncludeCallback includeCallback)
+        {
+            HoconIncludeCallbackAsync wrappedIncludeCallback = null;
+            if (includeCallback != null)
+                wrappedIncludeCallback = (t, s) => Task.FromResult(includeCallback(t, s));
+
+            return AsyncHelper.RunSync(() => ParseTextAsync(text, resolveSubstitutions, wrappedIncludeCallback));
+        }
+
+        private async Task<HoconRoot> ParseTextAsync(string text, bool resolveSubstitutions, HoconIncludeCallbackAsync includeCallback)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+                throw new HoconParserException(
+                    $"Parameter {nameof(text)} is null or empty.\n" +
+                    "If you want to create an empty Hocon HoconRoot, use \"{}\" or just use \"new HoconRoot();\" instead.");
+
+            // Workaround for annoying end of line difference between windows and unix
+            if (Environment.NewLine == "\r\n")
+                text = text.Replace(Environment.NewLine, "\n");
```

Comment:
I'm not sure here - this seem to be important only during tokenization phase (where it should be easily a check for one or another), while this one line can cause entire `text` string to reallocate (akka.net default config has 42kB and it's not the only hocon config loaded) just for this single reason.

Reply:
Thanks for pointing that out, it was a leftover code from the previous code base, end of line are actually handled in the tokenizer now.